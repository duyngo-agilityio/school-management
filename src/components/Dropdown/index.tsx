'use client';

import { ChangeEvent, memo } from 'react';
import isEqual from 'react-fast-compare';

// Components
import { Select, SelectProps } from '@chakra-ui/react';

export interface Option {
  value: string;
  label: string;
}

interface DropdownProps extends SelectProps {
  options: Option[];
  placeholder?: string;
  width?: string;
  onChangeValue: (value: string) => void;
}

const Dropdown = ({
  options,
  placeholder = '',
  width,
  onChangeValue,
  value,
  ...rest
}: DropdownProps) => {
  const handleSelect = (e: ChangeEvent<HTMLSelectElement>) => {
    onChangeValue(e.target.value);
  };

  return (
    <Select
      w={width}
      placeholder={placeholder}
      onChange={handleSelect}
      value={value}
      {...rest}
    >
      {options.map(({ value = '', label = '' }) => (
        <option key={value} value={value}>
          {label}
        </option>
      ))}
    </Select>
  );
};

export default memo(Dropdown, isEqual);
