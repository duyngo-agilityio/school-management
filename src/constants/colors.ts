const bluePallet = {
  zodiac: '#152259',
  koamaru: '#13296A',
  havelock: '#1C5A8C',
  cornflower: '#3182ce',
  solitude: '#EBF6FF',
  catSkill: '#E2E8F0',
  linkWater: '#EFF3FA',
};

const grayPallet = {
  cod: '#1A1A1A',
  tundora: '#424242',
  emperor: '#4F4F4F',
  ash: '#8A8A8A',
  slate: '#595959',
  aluminum: '#BDBDBD',
};

const whitePallet = {
  seashell: '#F1F1F1',
  snow: '#FCFAFA',
  pure: '#FFFFFF',
};

const redPallet = {
  roman: '#DB5050',
};

const yellowPallet = {
  anzac: '#DBBD50',
};

export const customColors = {
  ...bluePallet,
  ...whitePallet,
  ...grayPallet,
  ...redPallet,
  ...yellowPallet,
} as const;
