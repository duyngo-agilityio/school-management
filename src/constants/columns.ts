const COLUMNS = {
  FIELDS: {
    NAME: 'fullName',
    ID: 'id',
    EMAIL: 'email',
    CLASS: 'class',
    GENDER: 'gender',
    SUBJECT: 'subject',
  },
  HEADER_NAME: {
    NAME: 'Name',
    SUBJECT: 'Subject',
    CLASS: 'Class',
    EMAIL: 'Email Address',
    GENDER: 'Gender',
    STUDENT_ID: 'Student ID',
  },
};

export const COLUMNS_STUDENT = [
  {
    field: COLUMNS.FIELDS.NAME,
    headerName: COLUMNS.HEADER_NAME.NAME,
  },
  {
    field: COLUMNS.FIELDS.ID,
    headerName: COLUMNS.HEADER_NAME.STUDENT_ID,
  },
  {
    field: COLUMNS.FIELDS.EMAIL,
    headerName: COLUMNS.HEADER_NAME.EMAIL,
  },
  {
    field: COLUMNS.FIELDS.CLASS,
    headerName: COLUMNS.HEADER_NAME.CLASS,
  },
  {
    field: COLUMNS.FIELDS.GENDER,
    headerName: COLUMNS.HEADER_NAME.GENDER,
  },
];

export const COLUMNS_TEACHER = [
  {
    field: COLUMNS.FIELDS.NAME,
    headerName: COLUMNS.HEADER_NAME.NAME,
  },
  {
    field: COLUMNS.FIELDS.SUBJECT,
    headerName: COLUMNS.HEADER_NAME.SUBJECT,
  },
  {
    field: COLUMNS.FIELDS.CLASS,
    headerName: COLUMNS.HEADER_NAME.CLASS,
  },
  {
    field: COLUMNS.FIELDS.EMAIL,
    headerName: COLUMNS.HEADER_NAME.EMAIL,
  },
  {
    field: COLUMNS.FIELDS.GENDER,
    headerName: COLUMNS.HEADER_NAME.GENDER,
  },
];
