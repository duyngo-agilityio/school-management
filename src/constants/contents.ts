export const NOTFOUND = {
  STUDENT: {
    TITLE: 'No Students at this time',
    DESCRIPTION: 'Students will appear here after they enroll in your school.',
    BACK_BUTTON: 'Back to Student List'
  },
  TEACHER: {
    TITLE: 'No Teachers at this time',
    DESCRIPTION: 'Teachers will appear here after they enroll in your school.',
    BACK_BUTTON: 'Back to Teacher List'
  },
};
