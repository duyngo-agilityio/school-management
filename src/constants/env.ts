export const BASE_URL = process.env.NEXT_PUBLIC_BASE_URL!;
export const IMG_KEY = process.env.NEXT_PUBLIC_IMG_KEY;
