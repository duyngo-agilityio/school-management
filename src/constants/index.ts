export * from './path';
export * from './placeholder';
export * from './colors';
export * from './columns';
export * from './messages';
export * from './options';
export * from './regex';
export * from './contents';
export * from './routes';
