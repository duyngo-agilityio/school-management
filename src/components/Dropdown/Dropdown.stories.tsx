// Dropdown.stories.ts|tsx

import type { Meta, StoryObj } from '@storybook/react';

// Constants
import { INPUT_PLACEHOLDER, OPTIONS_SUBJECT } from '@/constants';

// Components
import Dropdown from '@/components/Dropdown';

import { fn } from '@storybook/test';

const meta: Meta<typeof Dropdown> = {
  title: 'components/Dropdown',
  component: Dropdown,
  tags: ['autodocs'],
  argTypes: {
    options: { description: 'array included option to select' },
    placeholder: {
      description:
        ' specifies a short hint that describes the expected value of an input field',
    },
    width: {
      description: 'set width input how to way flexible',
    },
    onChangeValue: {
      description: 'change value typing input',
    },
    value: {
      description: 'value input',
    },
  },
};

export default meta;
type Story = StoryObj<typeof Dropdown>;

export const Default: Story = {
  args: {
    options: OPTIONS_SUBJECT,
    placeholder: INPUT_PLACEHOLDER.TEACHER,
    width: '179px',
    onChangeValue: fn(),
  },
};
