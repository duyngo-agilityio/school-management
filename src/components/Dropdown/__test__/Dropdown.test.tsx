import { fireEvent, render } from '@testing-library/react';

// Components
import Dropdown from '..';

// Constants
import { INPUT_PLACEHOLDER, OPTIONS_SUBJECT } from '@/constants';

const DropdownProps = {
  options: OPTIONS_SUBJECT,
  placeholder: INPUT_PLACEHOLDER.TEACHER,
  width: '',
  onChangeValue: jest.fn(),
};

const DropdownComponent = () => render(<Dropdown {...DropdownProps} />);

describe('Dropdown components', () => {
  it('should match snapshot for Dropdown', () => {
    const { container } = DropdownComponent();

    expect(container).toMatchSnapshot();
  });

  it('calls onChangeValue when an option is selected', () => {
    const { getByRole } = DropdownComponent();

    fireEvent.change(getByRole('combobox'), { target: { value: 'math' } });

    expect(DropdownProps.onChangeValue).toHaveBeenCalled();
  });
});
