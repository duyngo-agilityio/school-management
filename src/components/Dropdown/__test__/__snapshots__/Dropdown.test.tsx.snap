// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Dropdown components should match snapshot for Dropdown 1`] = `
<div>
  <div
    class="chakra-select__wrapper css-11680z6"
  >
    <select
      class="chakra-select css-1ik61og"
    >
      <option
        value=""
      >
        Search for a teacher by name or email
      </option>
      <option
        value="math"
      >
        Math
      </option>
      <option
        value="physical"
      >
        Physical
      </option>
      <option
        value="biology"
      >
        Biology
      </option>
    </select>
    <div
      class="chakra-select__icon-wrapper css-162mkon"
    >
      <svg
        aria-hidden="true"
        class="chakra-select__icon"
        focusable="false"
        role="presentation"
        style="width: 1em; height: 1em; color: currentColor;"
        viewBox="0 0 24 24"
      >
        <path
          d="M16.59 8.59L12 13.17 7.41 8.59 6 10l6 6 6-6z"
          fill="currentColor"
        />
      </svg>
    </div>
  </div>
</div>
`;
